lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -137,57 +162,159 @@ static LogicalResult createPoolingOp(
                              ValueRange{paddedInput, windowTensor},
                              outTensorInitialized, stridesAttr, dilationAttr)
                .getResult(0);
-
   return success();
 }
 
-
 namespace {
-class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
+template <typename OpTy, int Dim>
+class ConvertAtenMaxPoolOp : public OpConversionPattern<OpTy> {
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+
+private:
+  LogicalResult writePoolingMax3D(AtenMaxPool3dOp &op,
+                                  typename OpTy::Adaptor &adaptor,
```

Comment:
nit: I think `adaptor` is unsused

Reply:
`self` and `elementType` are actually derived from adaptor. So, I will keep that.

Reply:
Ah, missed that. Makes sense to keep it.